src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -50,5 +49,13 @@ public SNIMarsConnection GetConnection(SNIHandle lowerHandle)
         {
             return _connections[lowerHandle];
         }
+
+        /// <summary>
+        /// Clears all Mars connections managed by this class.
+        /// </summary>
+        public void ClearMarsConnections()
+        {
+            _connections.Clear();
```

Comment:
We've had some issues with dangling async callbacks after the connections/queries finish, so it might be realted.